riverdata/components/RDStateSites.js

```diff
@@ -1,24 +1,95 @@
 // RDStateSites.js
-import React, { Component } from 'react';
-import { Button, View, Text } from 'react-native';
+import React, { Component, useEffect, useState } from 'react';
+import { Button, View, Text, ScrollView, StatusBar } from 'react-native';
+import { NavigationActions } from "react-navigation";
 import { NavigationContainer } from '@react-navigation/native';
 import { createNativeStackNavigator } from '@react-navigation/native-stack';
-
+import Item from './Item';
 
 const RDStateSitesScreen = ({ route, navigation }) => {
-  return (
-  <>  
-    <Text>The state paramenter passed in is: {route.params.name}. 
-    This is where we will make an asynchronous API call to get the sites for the chosen stated.
-    </Text>
-    <Button
-          title="Go to selected site gauge list"
-          onPress={() =>
-            navigation.navigate('Site Gauges', { gaugeId: '123' })
+  const [data, setData] = useState({value: {timeSeries: []}});
+  const [isLoading, setIsLoading] = useState(false);
+  const [err, setErr] = useState('');
+  const [sites, setSites] = useState([]);
+
+  const handleClick = async () => {
+    setIsLoading(true);
+
+    try {
+      const response = await fetch('https://waterservices.usgs.gov/nwis/iv?format=json&stateCd=nj', {
+        method: 'GET',
+        headers: {
+          Accept: 'application/json',
+        },
+      });
+
+      if (!response.ok) {
+        throw new Error('Error! status: ' + response.status);
+      }
+
+      const result = await response.json();
+
+      console.log('result is: ', JSON.stringify(result, null, 4));
+      console.log("check1")
+      setData(result);
+      console.log("check2")
+    } catch (err) {
+      setErr(err.message);
+    } finally {
+      setIsLoading(false);
+      console.log("check3")
```

Comment:
you will want to remove your console.log statements in the future. TIP: Look for a solution that only logs when debugging